golem_messages/message.py

```diff
@@ -1,16 +1,37 @@
 import cbor2
+import datetime
 import hashlib
 import logging
 import pytz
 import struct
 import time
 from typing import Optional
 
+from . import exceptions
 from . import serializer
+from . import settings
 
 logger = logging.getLogger('golem.network.transport.message')
 
 
+def verify_time(msg):

```

Comment:
Maybe instead of passing whole message, just pass timestamp?
With this you could check it before constructing message object in deserialize.

Reply:
And you could test it without creating message objects.